### ISSUE TITLE
Non-workspace dependent tasks are ran multiple times

### ISSUE THREAD

**Issue by aklitzke** [sagiegurari/cargo-make#339]
**Features Description**
Let's say we have a workspace makefile:
```
[env]
CARGO_MAKE_EXTEND_WORKSPACE_MAKEFILE = "true"

[tasks.build]
dependencies = ["want-to-do-once"]

[tasks.want-to-do-once]
workspace = false
```

There doesn't seem to be a good way to make `want-to-do-once` run only once at the beginning of the build.

**Describe the solution you'd like**
I'll leave that up to you, but seems like a pretty common case. In particular I'm using it to generate a random number which is shared workspace-wide.

**Comment by sagiegurari** [comment]
i think you can do the following

* your main task (workspace-build) will be
  * workspace=false so when running from cli it will not run it per member
  * have dependency on run-once that just sets up an env var with the number
  * setup the real build task name in the env
  * run-task to be defined as do-on-members which enables to run a flow per member (enables to switch to workspace mode basically)

i would write the toml but I'm answtfrom my phone :)

**Comment by aklitzke** [comment]
Hmm, I was kinda hoping to accomplish this without defining a new task like `workspace-build`. There would be two technically two seemingly valid tasks that can be ran at the root level to build: `workspace-build` and `build`, and it's not immediately obvious to someone reading the code which to use IMO.

**Comment by sagiegurari** [comment]
in that case you could have a definition of workspace build and member build in root workspace. you can set them to private to prevent direct invocation.

then in workspace toml define build task that extends workspace-build and changes private to false and in each member also define a timl file with build task that extends member-build with private=false.

users will always run build but it will require a makefile for each member for this trick

**Comment by sagiegurari** [comment]
another way, if you create workspace and member build tasks as i wrote before, you can define a build task in workspace that does runtask with routing based if its currently in workspace or not via env var.
that way you don't have to create makefiles for members.

**Comment by sagiegurari** [comment]
@aklitzke did this resolve your issue?

**Comment by sagiegurari** [comment]
closing due to lack of feedback.
feel free to reopen if it didn't answer your need.
